### PR TITLE
Add data from controller to comments template

### PR DESCRIPTION
**Pull request by mejta**
- Right now comments template is rendered without data from the controller. This pull request fixes that.
- Filter that renders template should have PHP_INT_MAX priority.

### CODE REVIEW

File: app/filters.php [hunk 1]
```diff
@@ -72,15 +72,19 @@
...
 
     if ($theme_template) {
-        echo template($theme_template);
+        echo template($theme_template, $data);
         return get_stylesheet_directory().'/index.php';
     }
 
     return $comments_template;
-}, 100);
+}, PHP_INT_MAX);
```
Comment (retlehs) [inline app/filters.php]: why is this needed?

Comment (mejta) [reply app/filters.php]: After render there should not been anything else. If there is 100 some other plugin could modify the path with priority e.g. 200 and that would result in rendering comments twice. When we have render with PHP_INT_MAX, there is a small possibility that other plugin will modify the path after render.

Comment (mejta) [reply app/filters.php]: After render there should not been anything else. If there is 100 some other plugin could modify the path with priority e.g. 200 and that would result in rendering comments twice. When we have render with PHP_INT_MAX, there is a small possibility that other plugin will modify the path after render.

Comment (mejta) [reply app/filters.php]: @retlehs Should I keep `PHP_INT_MAX` in here?

File: app/filters.php [hunk 2]
```diff
@@ -72,10 +72,14 @@
         $comments_template
     );
 
+    $data = collect(get_body_class())->reduce(function ($data, $class) use ($template) {
```
Comment (mmirus) [inline app/filters.php]: I think `$template` is undefined in this scope. Should be `$comments_template`?

Comment (mejta) [reply app/filters.php]: @mmirus You are right, it's fixed.

File: app/filters.php (hunk 1 above)
Comment (retlehs) [reply app/filters.php]: we don't want to use `PHP_INT_MAX` in the theme /cc @QWp6t

Comment (mtx-z) [reply app/filters.php]: What is the reason you don't want to use `PHP_INT_MAX` (for my own information) ? . Thanks.

Comment (QWp6t) [reply app/filters.php]: I'm against it because the use of it is too hamfisted and unnecessary. I don't see a reason for us to be that aggressive with the use of that filter or any others.

Comment (mejta) [reply app/filters.php]: Ok, it's reverted. But generally, I think that after render there should be no chance for a hook with higher priority to render comments again. It's a very same situation like here (https://github.com/roots/sage/blob/master/app/filters.php#L63) where is also render in the filter and `PHP_INT_MAX` priority. I would suggest a consistent approach for both situations, but it's not a scope of this PR and should be in different PR.
